libs/execution_context/ebpf_maps.c

```diff
@@ -972,10 +1116,24 @@ ebpf_map_update_entry_with_handle(
 ebpf_result_t
 ebpf_map_delete_entry(_In_ ebpf_map_t* map, size_t key_size, _In_reads_(key_size) const uint8_t* key, int flags)
 {
+    uint32_t cpu_id = 0;
+    bool key_found = false;
+
     if (!(flags & EBPF_MAP_FLAG_HELPER) && (key_size != map->ebpf_map_definition.key_size)) {
         return EBPF_INVALID_ARGUMENT;
     }
-    return ebpf_map_function_tables[map->ebpf_map_definition.type].delete_entry(map, key);
+
+    // Erase the value from all per-cpu maps
```

Comment:
Why would you ever need to erase a value from all per-cpu maps?  If I understand right, this is just trying to optimize for the "next key" path called from UM.  I don't think that's a hot path.   Inserting and deleting entries from within an ebpf program is, however, a hot path, and should be doable without taking any locks.

Reply:
It's a requirement of bpf_delete_elem.

Reply:
This comment is out of date.